In [ ]:
# loading pandas and numpy libraries

import pandas as pd
import numpy as np


In [ ]:
# load cells_geo dataset
df_geo = pd.read_csv('cells_geo.csv', sep=';')
df_geo

Unnamed: 0    VILLES      STATUS  ...          AREA     CELL_ID   SITE_CODE
0              0    ADJAME  In Service  ...           NaN  fbb2e75af3  0bb492ef17
1              1    ADJAME  In Service  ...           NaN  1502501d78  0bb492ef17
2              2    ADJAME  In Service  ...           NaN  0c2e182a0c  0bb492ef17
3              3   ASSINIE  In Service  ...           NaN  69bdfc69fd  8a1e6ce31f
4              4   ASSINIE  In Service  ...           NaN  7a1029d2ab  8a1e6ce31f
...          ...       ...         ...  ...           ...         ...         ...
3969        3969  ZUENOULA  In Service  ...  CENTRE NORTH  1cfc2d7ab1  f91ed8d2e9
3970        3970  ZUENOULA  In Service  ...  CENTRE NORTH  3ae1708153  f91ed8d2e9
3971        3971  ZUENOULA  In Service  ...  CENTRE NORTH  cb71f8a29c  61bee9c70c
3972        3972  ZUENOULA  In Service  ...  CENTRE NORTH  d4a6a0b645  61bee9c70c
3973        3973  ZUENOULA  In Service  ...  CENTRE NORTH  f9f3e42589  61bee9c70c

[3974 rows x 12 columns]

In [ ]:
#Importing Excel from pandas
from pandas import ExcelWriter
from pandas import ExcelFile

#load cells_geo description dataset
df_geo_d = pd.read_excel('/content/cells_geo_description.xlsx', sheet_name=0)
df_geo_d

Column name                                        Description  Format
0         VILLES                                               City  String
1         STATUS                                  In Service or not  String
2   LOCALISATION                                  in ABIDJAN or not  String
3     DECOUPZONE                                  Geographical Zone  String
4       ZONENAME                                       Name of Zone  String
5      LONGITUDE                                          Longitude   Float
6       LATITUDE                                           Latitude   Float
7         REGION                                             Region  String
8           AREA                                               Area  String
9        CELL_ID                                     ID of the cell  String
10     SITE_CODE  Site (there are several cells per site, severa...  String

In [ ]:
# load CDR_description dataset
df_CDR = pd.read_excel('CDR_description.xlsx', sheet_name=0)
df_CDR

Column name                                        Description   Format
0          PRODUCT                                       Voice or SMS   String
1            VALUE                                      Billing price  Integer
2        DATE_TIME               Time in format yyyy-MM-dd hh:mm:ss.0   String
3     CELL_ON_SITE  Which cell in the site was used (not needed here)  Integer
4  DW_A_NUMBER_INT  Anonymized phone number of the person for whic...   String
5  DW_B_NUMBER_INT        Anonymized phone number of the counterparty   String
6        COUNTRY_A                  Country of party A (useless here)   String
7        COUNTRY_B                  Country of party B (useless here)   String
8          CELL_ID                                     ID of the cell   String
9          SITE_ID                                     ID of the SITE   String

In [ ]:
#load Telcom Dataset 1

df1 = pd.read_csv('http://bit.ly/Telcom_dataset1')
df1

PRODUTC  VALUE               DATETIME  ... COUNTRY_B     CELL_ID     SITE_ID
0      Voice      0  2012-05-06 23:04:37.0  ...     19e2e  /TJNe+Mmtu  /TJNe+Mmtu
1        sms      0  2012-05-06 23:05:25.0  ...     19e2e  /+cKZKIp41  /+cKZKIp41
2        sms     25  2012-05-06 23:05:41.0  ...     19e2e  /u0FSD+ahi  /u0FSD+ahi
3      Voice     56  2012-05-06 23:05:42.0  ...     19e2e  +cNeJzsTp3  +cNeJzsTp3
4      Voice     14  2012-05-06 23:06:29.0  ...     19e2e  051KhYDCpv  051KhYDCpv
...      ...    ...                    ...  ...       ...         ...         ...
4996     sms    100  2012-05-07 00:01:54.0  ...     OTHER  0c2371d9bc  1e751d08a4
4997   Voice      1  2012-05-07 00:01:54.0  ...     19e2e  5fb791cf56  65efe2bc62
4998   Voice      0  2012-05-07 00:01:54.0  ...     19e2e  7773627c12  9ef9616350
4999   Voice      0  2012-05-07 00:01:54.0  ...     19e2e  3f4af49d6c  969b8d0e39
5000   Voice      7  2012-05-07 00:01:54.0  ...     19e2e  6f4d9b77a5  ebfc58af70

[5001 rows x 10 columns]

In [ ]:
#load Telcom Dataset 2

df2 = pd.read_csv('http://bit.ly/Telcom_dataset2')
df2

PRODUCT  VALUE              DATE_TIME  ... COUNTRY_B     CELL_ID     SITE_ID
0        sms      0  2012-05-07 23:02:06.0  ...     19e2e  +854AcBQT2  +854AcBQT2
1      Voice      0  2012-05-07 23:03:44.0  ...     19e2e  +854AcBQT2  +854AcBQT2
2        sms      0  2012-05-07 23:04:06.0  ...     19e2e  +laSrk7g6q  +laSrk7g6q
3      Voice     37  2012-05-07 23:04:48.0  ...     19e2e  /xvRrCVKoQ  /xvRrCVKoQ
4        sms     25  2012-05-07 23:04:58.0  ...     19e2e  +bO+qkSonO  +bO+qkSonO
...      ...    ...                    ...  ...       ...         ...         ...
4996     sms      0  2012-05-08 00:01:53.0  ...     19e2e  896459aace  b7d00e59e4
4997     sms     25  2012-05-08 00:01:53.0  ...     19e2e  7e90b35166  211015b14d
4998     sms      0  2012-05-08 00:01:53.0  ...     19e2e  fe70e431df  c7ea989264
4999    data      0  2012-05-08 00:01:53.0  ...     OTHER  ffa6759bb2         NaN
5000   Voice      0  2012-05-08 00:01:53.0  ...     19e2e  8630058aca  b47a998a3e

[5001 rows x 10 columns]

In [ ]:
#load telcom dataset 3
df3 = pd.read_csv('http://bit.ly/Telcom_dataset3')
df3

PRODUCT  VALUE              DATE_TIME  ... COUNTRY_B      CELLID     SIET_ID
0      Voice     61  2012-05-08 23:01:28.0  ...     19e2e  +38u2u/rfx  +38u2u/rfx
1        sms      0  2012-05-08 23:01:31.0  ...     19e2e  +Fs4mTvdKx  +Fs4mTvdKx
2      Voice     10  2012-05-08 23:02:59.0  ...     19e2e  +ZaaZozy/+  +ZaaZozy/+
3        sms      0  2012-05-08 23:03:04.0  ...     19e2e  +cNeJzsTp3  +cNeJzsTp3
4      Voice     22  2012-05-08 23:03:16.0  ...     19e2e  /eeMklgyJA  /eeMklgyJA
...      ...    ...                    ...  ...       ...         ...         ...
4996   Voice      0  2012-05-09 00:01:43.0  ...     19e2e  2aa04dcab7  f7d63a29e5
4997   Voice     62  2012-05-09 00:01:43.0  ...     19e2e  7f1f2f889f  b4ddb9e93c
4998   Voice     32  2012-05-09 00:01:43.0  ...     19e2e  a4731e780e  d68260ef50
4999   Voice     30  2012-05-09 00:01:43.0  ...     19e2e  98ed726da2  b4130b54ab
5000   Voice     60  2012-05-09 00:01:43.0  ...     19e2e  1c11a2fcbb  6d70dae094

[5001 rows x 10 columns]

In [ ]:
#selecting the most used cities

#Select cities that appear most times

#Cities to be grouped
geo_villes = df_geo.groupby('VILLES').count()
geo_villes


Unnamed: 0  STATUS  LOCALISATION  ...  AREA  CELL_ID  SITE_CODE
VILLES                                           ...                          
ABENGOUROU             21      21            21  ...    21       21         21
ABOBO                 255     250           255  ...   247      255        255
ABOISSO                18      18            18  ...    18       18         18
ABOISSO COMOE           3       3             3  ...     3        3          3
ABONGOUA                3       3             3  ...     3        3          3
...                   ...     ...           ...  ...   ...      ...        ...
ZATTA                   3       3             3  ...     3        3          3
ZIKISSO                 4       4             4  ...     4        4          4
ZON-HOUNIEN             3       3             3  ...     3        3          3
ZOUKOUGBEU              3       3             3  ...     3        3          3
ZUENOULA                9       9             9  ...     9        9          9

[504 rows x 11 columns]

In [ ]:
#sorting the cities in DESC order to get the cities with highest activity

geo_villes.sort_values(by=['STATUS'], ascending=False).head(20)

Unnamed: 0  STATUS  LOCALISATION  ...  AREA  CELL_ID  SITE_CODE
VILLES                                          ...                          
COCODY               354     354           354  ...   354      354        354
YOPOUGON             295     289           295  ...   295      295        295
ABOBO                255     250           255  ...   247      255        255
MARCORY              141     141           141  ...   141      141        141
TREICHVILLE          132     132           132  ...   132      132        132
ADJAME               110     109           110  ...   105      110        110
YAMOUSSOUKRO         104     104           104  ...   104      104        104
PLATEAU              103     103           103  ...   100      103        103
KOUMASSI              91      91            91  ...    91       91         91
BOUAKE                65      65            65  ...    65       65         65
PORT-BOUET            61      59            61  ...    61       61         61
DALOA                 59      59            59  ...    59       59         59
SAN-PEDRO             55      55            55  ...    55       55         55
GAGNOA                55      55            55  ...    55       55         55
ATTECOUBE             46      45            46  ...    46       46         46
DIVO                  39      36            39  ...    39       39         39
KORHOGO               36      32            36  ...    36       36         36
BINGERVILLE           30      30            30  ...    27       30         30
SOUBRE                28      28            28  ...    28       28         28
GRAND-BASSAM          27      27            27  ...    27       27         27

[20 rows x 11 columns]

In the above findings, the most used cities are :
1. COCODY - 354
2. Yopuogon - 295
3. ABOBO - 255
4. MARCORY - 141
5. TRETCHVILLE - 132

In [ ]:
#Grouping localization into comms that occurred in abidjan, and thode that did not

localize = df_geo.groupby(['LOCALISATION']).count()
localize

Unnamed: 0  VILLES  STATUS  ...  AREA  CELL_ID  SITE_CODE
LOCALISATION                              ...                          
ABIDJAN             1669    1669    1636  ...  1650     1669       1669
INTERIEUR           2305    2305    2271  ...  2301     2305       2305

[2 rows x 11 columns]

Out of the 3974 records ;
1669 mobile comms occurred in ABIDJAN
2305 comms occurred in the INTERIUR

In [1]:
#gathering values of the calls

#day 1
day1_value = df1['VALUE'].sum()
day1_value

NameError: name 'df1' is not defined

In [ ]:
#day 2
day2_value = df2['VALUE'].sum()
day2_value 

121699

In [ ]:
#day 3
day3_value = df3['VALUE'].sum()
day3_value

132234

In [ ]:
#total values of the calls and sms

tot_value = day1_value + day2_value + day3_value
tot_value

383307

The total amount of money made by MTN in the three days is 383,307 XOF(CFA franc)

In [ ]:
#import datetime library
from datetime import datetime

In [ ]:
#combining the three days tables with cells_geo table to get cities that had most activity at given times of the day.

#Join the three tables with the geo table to get cities

#Table 3 

join_try1 = df3.join(df_geo, how='outer')
join_try1

#cleaning null values
nan = float("NaN")
join_try1.replace("", nan, inplace=True)
join_try1.dropna()

PRODUCT  VALUE  ...     CELL_ID   SITE_CODE
23       sms     25  ...  f740406877  0f34c36ac5
24       sms      0  ...  959e54c8f7  0f34c36ac5
25     Voice      0  ...  56b7e7000c  0f34c36ac5
26       sms     25  ...  bc51c17a6f  0f34c36ac5
27      data      1  ...  e460975c9f  0f34c36ac5
...      ...    ...  ...         ...         ...
3969     sms      0  ...  1cfc2d7ab1  f91ed8d2e9
3970     sms      0  ...  3ae1708153  f91ed8d2e9
3971     sms     25  ...  cb71f8a29c  61bee9c70c
3972     sms     25  ...  d4a6a0b645  61bee9c70c
3973     sms      0  ...  f9f3e42589  61bee9c70c

[3412 rows x 22 columns]

In [ ]:

#remane cell id column
df2_new = df2.rename(columns={'CELL_ID': 'CELLID'}, inplace=True)

#Table 2 merging
join_try2 = df2.join(df_geo, how='outer')
join_try2
#cleaning null values
nan = float("NaN")
join_try2.replace("", nan, inplace=True)
join_try2.dropna()

PRODUCT  VALUE  ...     CELL_ID   SITE_CODE
23       sms      0  ...  f740406877  0f34c36ac5
24       sms      0  ...  959e54c8f7  0f34c36ac5
25     Voice      0  ...  56b7e7000c  0f34c36ac5
26       sms      0  ...  bc51c17a6f  0f34c36ac5
27       sms      0  ...  e460975c9f  0f34c36ac5
...      ...    ...  ...         ...         ...
3968     sms      0  ...  dcd3a41af8  f91ed8d2e9
3969     sms      0  ...  1cfc2d7ab1  f91ed8d2e9
3970     sms      0  ...  3ae1708153  f91ed8d2e9
3972   Voice     34  ...  d4a6a0b645  61bee9c70c
3973   Voice      0  ...  f9f3e42589  61bee9c70c

[3381 rows x 22 columns]

In [ ]:
#renaming cell id column 
df1.rename(columns={'CELL_ID': 'CELLID'}, inplace=True)
#Table 1 

join_try3 = df1.join(df_geo, how='outer')
join_try3

#cleaning null values
nan = float("NaN")
join_try3.replace("", nan, inplace=True)
join_try3.dropna()

PRODUTC  VALUE  ...     CELL_ID   SITE_CODE
23      data      0  ...  f740406877  0f34c36ac5
24     Voice      0  ...  959e54c8f7  0f34c36ac5
25      data      0  ...  56b7e7000c  0f34c36ac5
26     Voice      0  ...  bc51c17a6f  0f34c36ac5
27     Voice     19  ...  e460975c9f  0f34c36ac5
...      ...    ...  ...         ...         ...
3969   Voice     20  ...  1cfc2d7ab1  f91ed8d2e9
3970   Voice     30  ...  3ae1708153  f91ed8d2e9
3971   Voice      0  ...  cb71f8a29c  61bee9c70c
3972   Voice     29  ...  d4a6a0b645  61bee9c70c
3973   Voice     40  ...  f9f3e42589  61bee9c70c

[3329 rows x 22 columns]

In [ ]:
# Business hours is assumed to be 08:00:00 until 18:00:00

#table1
biz_df1 = joined_df1.loc[(joined_df1['DATETIME'] >= '2012-05-06 08:00:00.0') & (joined_df1['DATETIME'] <= '2012-05-06 18:00:00.0') ]
biz_df1

Unnamed: 0    VILLES      STATUS  ... COUNTRY_B     CELL_ID     SITE_ID
0              0    ADJAME  In Service  ...     19e2e  /TJNe+Mmtu  /TJNe+Mmtu
1              1    ADJAME  In Service  ...     19e2e  /+cKZKIp41  /+cKZKIp41
2              2    ADJAME  In Service  ...     19e2e  /u0FSD+ahi  /u0FSD+ahi
3              3   ASSINIE  In Service  ...     19e2e  +cNeJzsTp3  +cNeJzsTp3
4              4   ASSINIE  In Service  ...     19e2e  051KhYDCpv  051KhYDCpv
...          ...       ...         ...  ...       ...         ...         ...
3969        3969  ZUENOULA  In Service  ...     19e2e  07ed226112  334fdb9912
3970        3970  ZUENOULA  In Service  ...     19e2e  729b6b4b22  9411ac7df8
3971        3971  ZUENOULA  In Service  ...     19e2e  77255d2612  c6400e443f
3972        3972  ZUENOULA  In Service  ...     19e2e  706d8914d0  22389546ec
3973        3973  ZUENOULA  In Service  ...     19e2e  3424273a07  ae6cdb80c1

[3974 rows x 22 columns]

In [ ]:
#most used city for three days

#day 3
day3_city = join_try1.groupby('VILLES').count()
day3_city

#sorting
day3_city.sort_values(by=['PRODUCT'], ascending=False).head(20)

PRODUCT  VALUE  DATE_TIME  ...  AREA  CELL_ID  SITE_CODE
VILLES                                   ...                          
COCODY            354    354        354  ...   354      354        354
YOPOUGON          295    295        295  ...   295      295        295
ABOBO             255    255        255  ...   247      255        255
MARCORY           141    141        141  ...   141      141        141
TREICHVILLE       132    132        132  ...   132      132        132
ADJAME            110    110        110  ...   105      110        110
YAMOUSSOUKRO      104    104        104  ...   104      104        104
PLATEAU           103    103        103  ...   100      103        103
KOUMASSI           91     91         91  ...    91       91         91
BOUAKE             65     65         65  ...    65       65         65
PORT-BOUET         61     61         61  ...    61       61         61
DALOA              59     59         59  ...    59       59         59
SAN-PEDRO          55     55         55  ...    55       55         55
GAGNOA             55     55         55  ...    55       55         55
ATTECOUBE          46     46         46  ...    46       46         46
BONGOUANOU         42     42         42  ...    42       42         42
DIVO               39     39         39  ...    39       39         39
KORHOGO            36     36         36  ...    36       36         36
BINGERVILLE        30     30         30  ...    27       30         30
SOUBRE             28     28         28  ...    28       28         28

[20 rows x 21 columns]

The most used city in day 3 was COCODY, 354 times


In [ ]:
#Day 2

day2_city = join_try2.groupby('VILLES').count()
day2_city

#sorting
day2_city.sort_values(by=['PRODUCT'], ascending=False).head(20)

PRODUCT  VALUE  DATE_TIME  ...  AREA  CELL_ID  SITE_CODE
VILLES                                   ...                          
COCODY            354    354        354  ...   354      354        354
YOPOUGON          295    295        295  ...   295      295        295
ABOBO             255    255        255  ...   247      255        255
MARCORY           141    141        141  ...   141      141        141
TREICHVILLE       132    132        132  ...   132      132        132
ADJAME            110    110        110  ...   105      110        110
YAMOUSSOUKRO      104    104        104  ...   104      104        104
PLATEAU           103    103        103  ...   100      103        103
KOUMASSI           91     91         91  ...    91       91         91
BOUAKE             65     65         65  ...    65       65         65
PORT-BOUET         61     61         61  ...    61       61         61
DALOA              59     59         59  ...    59       59         59
SAN-PEDRO          55     55         55  ...    55       55         55
GAGNOA             55     55         55  ...    55       55         55
ATTECOUBE          46     46         46  ...    46       46         46
BONGOUANOU         42     42         42  ...    42       42         42
DIVO               39     39         39  ...    39       39         39
KORHOGO            36     36         36  ...    36       36         36
BINGERVILLE        30     30         30  ...    27       30         30
SOUBRE             28     28         28  ...    28       28         28

[20 rows x 21 columns]

In [ ]:
#Data cleaning
#Renaming Produtc to Product
df1.rename(columns={'PRODUTC': 'PRODUCT'}, inplace=True)
#day 1
day1_city = join_try3.groupby('VILLES').count()
day1_city

#sorting
day1_city.sort_values(by=['VALUE'], ascending=False).head(20)

PRODUTC  VALUE  DATETIME  ...  AREA  CELL_ID  SITE_CODE
VILLES                                  ...                          
COCODY            354    354       354  ...   354      354        354
YOPOUGON          295    295       295  ...   295      295        295
ABOBO             255    255       255  ...   247      255        255
MARCORY           141    141       141  ...   141      141        141
TREICHVILLE       132    132       132  ...   132      132        132
ADJAME            110    110       110  ...   105      110        110
YAMOUSSOUKRO      104    104       104  ...   104      104        104
PLATEAU           103    103       103  ...   100      103        103
KOUMASSI           91     91        91  ...    91       91         91
BOUAKE             65     65        65  ...    65       65         65
PORT-BOUET         61     61        61  ...    61       61         61
DALOA              59     59        59  ...    59       59         59
SAN-PEDRO          55     55        55  ...    55       55         55
GAGNOA             55     55        55  ...    55       55         55
ATTECOUBE          46     46        46  ...    46       46         46
BONGOUANOU         42     42        42  ...    42       42         42
DIVO               39     39        39  ...    39       39         39
KORHOGO            36     36        36  ...    36       36         36
BINGERVILLE        30     30        30  ...    27       30         30
SOUBRE             28     28        28  ...    28       28         28

[20 rows x 21 columns]